app/models/comment.rb

```diff
@@ -24,6 +24,19 @@ class Comment < ApplicationRecord
 
   delegate :user, to: :pseud, allow_nil: true
 
+  validate :can_be_guest_reply, if: :reply_comment?, unless: :pseud_id, on: :create
+  # Whether the writer of the comment this is replying to allows guest replies
+  def can_be_guest_reply
```

Comment:
For consistency, can you put the comment above the `validate` line, like the `not_blocked` validations do?

I don't think it's super important for the methods to have very distinct names, since they're pretty closely connected -- in fact, we have both a `check_for_spam?` method and a `check_for_spam` validation in this model. But if we wanted to tweak the name, `guest_can_reply` would be in line with the `guest_cannot_kudos_restricted_work` kudos validation.